runner.go

```diff
@@ -741,6 +822,125 @@ func (r *Runner) deletePid() error {
 	return nil
 }
 
+// reloadChild sends the reload signal to the child process.
+func (r *Runner) reloadChild() error {
+	log.Printf("[INFO] (runner) reloading child process")
+
+	if r.config.Exec.Splay > 0 {
+		ns := r.config.Exec.Splay.Nanoseconds()
+		offset := rand.Int63n(ns)
+
+		log.Printf("[DEBUG] (runner) waiting %.2fs for random splay",
+			time.Duration(offset).Seconds())
+
+		select {
+		case <-time.After(time.Duration(offset)):
+		case <-r.DoneCh:
+		}
+	}
+
+	return r.cmd.Process.Signal(r.config.Exec.ReloadSignal)
+}
+
+// killChild kills the attached process as part of a graceful shutdown. This
+// function should only be called if there is an attached and running process.
+func (r *Runner) killChild() {
+	log.Printf("[INFO] (runner) killing child process")
+
+	exited := false
+
+	if r.config.Exec.Splay > 0 {
+		ns := r.config.Exec.Splay.Nanoseconds()
+		offset := rand.Int63n(ns)
+
+		log.Printf("[DEBUG] (runner) waiting %.2fs for random splay",
+			time.Duration(offset).Seconds())
+
+		select {
+		case <-time.After(time.Duration(offset)):
+		case <-r.DoneCh:
+		}
+	}
+
+	if err := r.cmd.Process.Signal(r.config.Exec.KillSignal); err == nil {
```

Comment:
So on windows the only supported signal you can send is a SIGKILL. Not sure if you want to guard against that. Probably when you support restart mode, only enable that on windows?